controllers/paddlejob_helper.go

```diff
@@ -369,3 +376,97 @@ func constructService4Pod(pod corev1.Pod) *corev1.Service {
 	}
 	return svc
 }
+
+// for volcano
+
+func withVolcano(pdj *pdv1.PaddleJob) bool {
+	check := func(rs *pdv1.ResourceSpec) bool {
+		if rs != nil && rs.Template.Spec.SchedulerName == schedulerNameVolcano {
+			return true
+		} else {
+			return false
+		}
+	}
+	if check(pdj.Spec.PS) || check(pdj.Spec.Worker) || check(pdj.Spec.Heter) {
+		return true
+	} else {
+		return false
+	}
+}
+
+func constructPodGroup(pdj *pdv1.PaddleJob) *volcano.PodGroup {
+	pg := &volcano.PodGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: pdj.Namespace,
+			Name:      pdj.Name,
+		},
+	}
+
+	if pdj.Spec.SchedulingPolicy != nil {
+		// minAvailable specified by user which not equals to total replicas
+		// DO NOT make sense in current paddle scenario
+		if pdj.Spec.SchedulingPolicy.MinAvailable != nil {
+			pg.Spec.MinMember = *pdj.Spec.SchedulingPolicy.MinAvailable
+		}
+		if pdj.Spec.SchedulingPolicy.Queue != "" {
+			pg.Spec.Queue = pdj.Spec.SchedulingPolicy.Queue
+		}
+		if pdj.Spec.SchedulingPolicy.PriorityClass != "" {
+			pg.Spec.PriorityClassName = pdj.Spec.SchedulingPolicy.PriorityClass
+		}
+		if pdj.Spec.SchedulingPolicy.MinResources != nil {
+			pg.Spec.MinResources = &pdj.Spec.SchedulingPolicy.MinResources
+		} else {
+			pg.Spec.MinResources = getPGMinResource(pdj)
+		}
+	} else {
+		pg.Spec.MinMember = getTotalReplicas(pdj)
+		pg.Spec.MinResources = getPGMinResource(pdj)
+	}
+
+	return pg
+}
+
+func getTotalReplicas(pdj *pdv1.PaddleJob) int32 {
+	count := func(rs *pdv1.ResourceSpec) int32 {
+		if rs != nil {
+			return int32(rs.Replicas)
+		} else {
+			return 0
+		}
+	}
+	return count(pdj.Spec.PS) + count(pdj.Spec.Worker) + count(pdj.Spec.Heter)
+}
+
+func getPGMinResource(pdj *pdv1.PaddleJob) *corev1.ResourceList {
+	addRes := func(crr, res corev1.ResourceList) {
+		for name, quantity := range res {
+			if value, ok := crr[name]; !ok {
+				crr[name] = quantity.DeepCopy()
+			} else {
+				value.Add(quantity)
+				crr[name] = value
+			}
+		}
+	}
+	// consider only the case minMember == minAvailable
```

Comment:
why considering minAvailable = sumof(replicas) only?

Reply:
paddlejob only make sense when all pods got to run together in gang-scheduling context. maybe TODO for further supporting.

Reply:
okay.